### PR TITLE
Revamped repository: add file linting and links checking

### PR DESCRIPTION
**Pull request by gwarf**
Add file linting and links checking using GitHub actions.
Lint files.
Document some repository management rules

### CODE REVIEW

File: .github/CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ We welcome issues reported by the public, and more specifically the community
 of people using this repository.
 The EGI team is small and cannot support all requests equally.
 While we undertake to do everything in our power to respond to issues
-timely, and to prioritise issues based on reasonable requests from
+timeously, and to prioritise issues based on reasonable requests from
```
Comment (enolfc) [inline .github/CODE_OF_CONDUCT.md]: timeously seems not that common in English

Comment (gwarf) [reply .github/CODE_OF_CONDUCT.md]: So I merged this change from Code of conduct in some others repos we have, it's may be an artefact of some more local English :)
I was feeling timely should more have to be used here liked `in a timely manner`. Will ask english colleagues.

Comment (gwarf) [reply .github/CODE_OF_CONDUCT.md]: Addressed.

**Review by enolfc — APPROVED** [review]
LGTM!

**Pull request merged**